js/script.js

```diff
@@ -1,15 +1,215 @@
-// Javascript Code.
-$(document).ready(function () {
-  // $.get("_posts/2014-11-09.json", function(data){
-  //
-  //   var posts = JSON.parse(data);
-  //   $(".posts").html(posts['posts'][0]['title']);
-  //
-  // });
-
-  $(this).load("_posts/posts.json", function(data){
-    var posts = JSON.parse(data);
-    $(".posts").html(posts['posts'][0]['title']);
-    //
-  });
-});
+// Javascript Code.
+
+//This is the search box on the page
+$(document).ready(function(){
+    $('.buttonsearch').click(function() {
+        var toAdd = $('input[name=checkListItem]').val();   
+        $('.list').append('<div class="item">' + toAdd + '</div>')
+    });
+});
+
+//This is the search box on the page
+
+
+
+$(document).on('ready',function()
+{
+$(".pull-me").click(function(event){
+event.preventDefault();
+$(".panelsearch").show("slow");
+$(".pull-me").hide("slow");
+$(".hide_search").show("slow");
+});
+
+$(".hide_search").click(function(event){
+event.preventDefault();
+$(".pull-me").show(1000);
+$(".panelsearch").hide(1000);
+});
+});
+
+//This is the search box on the page!!!! 
+jQuery.fn.extend({
+    mark: function(search0, classCCSSsearch){
+        var regex = new RegExp("(<[^>]*>)|("+ search0.replace(/([-.*+?^${}()|[\]\/\\])/g,"\\$1") +')', 'ig');
+        var newHtml=this.html(this.html().replace(regex, function(a, b, c){
+            return (a.charAt(0) == "<") ? a : "<span class=\""+ classCCSSsearch +"\">" + a + "</span>";
+        }));
+        return newHtml;
+    }
+});
+//To highlight text
+function highlightText(){
+    $('.search1').mark(textbox.value,"highlight");
+    $('.search2').mark(textbox.value,"highlight");
+    $('.search3').mark(textbox.value,"highlight");
+    $('.search4').mark(textbox.value,"highlight");
+    $('.search5').mark(textbox.value,"highlight");
+    $('.search6').mark(textbox.value,"highlight");
+    $('.search7').mark(textbox.value,"highlight");
+    $('.search8').mark(textbox.value,"highlight");
+    $('.search9').mark(textbox.value,"highlight");
+    $('.search10').mark(textbox.value,"highlight");
+    $('.search11').mark(textbox.value,"highlight");
+    $('.search12').mark(textbox.value,"highlight");
+    $('.search13').mark(textbox.value,"highlight");
+    $('.search14').mark(textbox.value,"highlight");
+    $('.search15').mark(textbox.value,"highlight");
+    $('.search16').mark(textbox.value,"highlight");
+}
+//end tehe search box
+
+
+  $(this).load("_posts/posts.json", function(data){
+    var posts = JSON.parse(data);
+
+    //
+    function order_date (date1, date2) {
+      return new Date(date2.date).getTime() - new Date(date1.date).getTime();
+    }
+
+    posts['posts'].sort(order_date);
+});
+//Here are the Json
+$(document).ready(function () {
+  // $.get("_posts/2014-11-09.json", function(data){

```

Comment:
In general it’s a good idea to remove unused code or legacy code to reduce the number of lines in our files. This way our files are cleaner, and its easier to maintain our code base.